src/main/java/edu/kit/provideq/toolbox/api/SolveRouter.java

```diff
@@ -142,6 +135,146 @@ private Mono<ServerResponse> handleSolutionRouteForMetaSolver(MetaSolver<?, ?, ?
     });
   }
 
+  private void handleRouteDocumentation(MetaSolver<?, ?, ?> metaSolver, Builder ops) {
+    var problemType = metaSolver.getProblemType();
+    ops
+        .operationId(getSolveRouteForProblemType(problemType))
+        .tag(problemType.getId())
+        .description("Solves a " + problemType.getId() + " problem. To solve the problem, "
+            + "either the meta-solver will choose the best available solver,"
+            + "or a specific solver selected in the request will be used.")
+        .requestBody(requestBodyBuilder()
+                .content(getRequestContent(metaSolver))
+                .required(true))
+        .response(getResponseOk(metaSolver))
+        .response(getResponseNotFound());
+  }
+
+  private static org.springdoc.core.fn.builders.apiresponse.Builder getResponseOk(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return responseBuilder()
+            .responseCode(String.valueOf(HttpStatus.OK.value()))
+            .content(contentBuilder()
+                    .mediaType(APPLICATION_JSON_VALUE)
+                    .example(getExampleSolved(metaSolver))
+                    .example(getExampleInvalid(metaSolver))
+                    .array(arraySchemaBuilder().schema(
+                            schemaBuilder().implementation(Solution.class))));
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder getExampleSolved(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return getExampleOk(
+            "Solved",
+            metaSolver,
+            solution -> {
+              solution.setSolutionData("Solution data to solve the problem");
+              solution.complete();
+            });
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder getExampleInvalid(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return getExampleOk(
+            "Error",
+            metaSolver,
+            solution -> {
+              solution.setDebugData("Some error occurred");
```

Comment:
I think `INVALID` refers to invalid problems only

Reply:
We also call `abort()` when some internal error occurs. Or do you want me to change something specific?

Reply:
Yes, I think that would make sense. We can keep INVALID and add something like ERROR. And there should probably be an equivalent to abort() for this too.

Reply:
Done in https://github.com/ProvideQ/toolbox-server/pull/47